block/block.go

```diff
@@ -114,19 +116,16 @@ func (m *Manager) isHeightAlreadyApplied(blockHeight uint64) (bool, error) {
 }
 
 func (m *Manager) attemptApplyCachedBlocks() error {
-	m.retrieverMu.Lock()
```

Comment:
by removing this, we basically can't know for sure that the `expectedHeight` from the time we get it, till the time we try to apply it, is correct (as another method could apply a different block within this time).
This in turn would fail the validation, which currently raises an error. 

so we either need to gracefully handle this, or assure the `expectedHeight` can't be changed between getting it to try to apply the block.